src/peft/tuners/lora/gptq.py

```diff
@@ -53,9 +56,25 @@ def __init__(
             init_lora_weights=init_lora_weights,
             use_rslora=use_rslora,
             use_dora=use_dora,
+            use_qalora=use_qalora,
             lora_bias=lora_bias,
+            qalora_group_size=qalora_group_size,
         )
 
+    def resolve_lora_variant(self, *, use_dora: bool, use_qalora: bool, **kwargs) -> Optional[LoraVariant]:
+        if use_dora and use_qalora:
+            NotImplementedError
+        elif use_dora:
+            from .variants import DoraLinearVariant
+            variant = DoraLinearVariant()
+        elif use_qalora:
+            if self.in_features % kwargs["qalora_group_size"] == 0:
```

Comment:
Let's not perform the check here. Instead, let's move it inside `QALoraLinearVariant.init`.

Reply:
This check can now be removed, right?